src/main/java/com/conveyal/r5/profile/PerTargetPropagater.java

```diff
@@ -361,6 +361,9 @@ private void propagateTransit (int targetIndex, LinkedPointSet linkedTargets) {
                             perIterationTravelTimes[iteration] = timeAtTarget;
                             if (calculateComponents || targetIndex == destinationIndex) {
                                 Path[] pathsToStops = pathsToStopsForIteration.get(iteration);
+                                for (Path path : pathsToStops) {
+                                    path.egressMode = linkedTargets.streetMode;
+                                }
```

Comment:
Summarizing discussion in our call: the loop is not necessary, at this nesting level we only need to update the path to one stop at one iteration. Also, the pathsToStops are a pre-existing external input to this propagater object (and to this single-target propagation operation) so should ideally not be written to. The egress mode via a particular stop may change for every target, so if we write to that path we are essentially (ab)using the input path as a sort of scratch buffer for this single target; the best egress mode via a given stop could even conceivably be different for every iteration, in which case whatever mode was best via that stop on the final iteration would take precedence over all the other iterations at this target. It is probably provable that that never happens, but proper function is predicated on some big assumptions.

So, in order to keep the input paths in tact and to reflect the existing structure of the propagator object, we can store the egress modes for the best paths in a per-target scratch array in an instance method parallel to perIterationPaths (perhaps called perIterationEgressModes). This disposable array is then supplied to the pathWriter along with the access modes for this origin, and spliced onto the paths as they are written. The problem being that we sort the travel times and pick paths close to a certain percentile of travel time - we would need to somehow maintain the link between egress modes and those paths. So for now we can just use the input path as a per-iteration scratch object, but put a big warning on it that this only works when we are processing one target at a time and the egress modes on the paths are considered meaningless outside that context.